### PR TITLE
fix: [CDS-73277]: infinite scroll support in servicesList under  pipeline executions filters

### PR DESCRIPTION
**Pull request by satyam-harness**
#### Screenshots

NA

#### PR Checklist

<details>
<summary>Please check if your PR fulfils the following requirements</summary>

- [ ] Tests for the changes have been added. Ideally, include a test that fails without this PR but passes with it.
- [ ] Docs have been [added/updated](https://harness.atlassian.net/jira/software/c/projects/DOC/boards/40).
</details>

<details>
<summary>Use the following comments to re-trigger PR Checks</summary>

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Sonar: `retrigger sonar`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Feature Name Check: `trigger featurenamecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress Rest: `retrigger cypress-rest`
- Cypress CD: `retrigger cypress-cd`
- Cypress Pipeline: `retrigger cypress-pipeline`
- Cypress CV: `retrigger cypress-cv`
- Fix Prettier: `fix prettier`
</details>

### CODE REVIEW

**Review by vikrant-harness — APPROVED** [review]

File: src/modules/70-pipeline/pages/execution-list/ExecutionListFilterForm/__tests__/ExecutionListFilterForm.test.tsx [hunk 1]
```diff
@@ -33,6 +33,10 @@ const params = {
 
 const TEST_PATH = routes.toPipelines({ ...accountPathProps, ...pipelinePathProps, ...pipelineModuleParams })
 
+jest.mock('services/cd-ng', () => ({
+  getServiceListPromise: jest.fn().mockImplementation(() => Promise.resolve([]))
+}))
+
```
Comment (chetanvekariya-harness) [inline src/modules/70-pipeline/pages/execution-list/ExecutionListFilterForm/__tests__/ExecutionListFilterForm.test.tsx:39]: Can we add some basic test cases as well?

File: src/modules/70-pipeline/pages/execution-list/ExecutionListFilterForm/ExecutionListFilterForm.tsx [hunk 2]
```diff
@@ -134,11 +184,45 @@ export function ExecutionListFilterForm<
...
+          />
+        </div>
+      )
+    }
+
+    const itemListRenderer: ItemListRenderer<SelectOption> = itemListProps => (
+      <Menu>
+        {isEmptyContent ? (
+          <Layout.Vertical flex={{ align: 'center-center' }} width={'100%'} height={'100%'}>
+            <div>No service available</div>
```
Comment (chetanvekariya-harness) [inline src/modules/70-pipeline/pages/execution-list/ExecutionListFilterForm/ExecutionListFilterForm.tsx:212]: Move this one to strings

File: src/modules/70-pipeline/pages/execution-list/ExecutionListFilterForm/ExecutionListFilterForm.tsx [hunk 3]
```diff
@@ -42,7 +49,50 @@ export function ExecutionListFilterForm<
...
+      orgIdentifier,
+      projectIdentifier,
+      size: pageSize.current,
+      searchTerm,
+      includeAllServicesAccessibleAtScope: CDS_OrgAccountLevelServiceEnvEnvGroup
+    }),
+    [accountId, orgIdentifier, projectIdentifier, searchTerm]
+  )
+
+  const [isFetchingServiceNextTime, setIsFetchingServiceNextTime] = useState(true)
```
Comment (chetanvekariya-harness) [inline src/modules/70-pipeline/pages/execution-list/ExecutionListFilterForm/ExecutionListFilterForm.tsx:70]: nit: isFetchingMoreServices

File: src/modules/70-pipeline/pages/execution-list/ExecutionListFilterForm/ExecutionListFilterForm.tsx [hunk 4]
```diff
@@ -42,7 +49,50 @@ export function ExecutionListFilterForm<
 >(props: ExecutionListFilterFormProps<T>): React.ReactElement {
   const { getString } = useStrings()
   const { module } = useParams<ModulePathParams>()
+  const { accountId, projectIdentifier, orgIdentifier } = useParams<PipelineType<PipelinePathProps>>()
   const { type, formikProps, isCDEnabled, isCIEnabled, initialValues } = props
+  const { CDS_OrgAccountLevelServiceEnvEnvGroup } = useFeatureFlags()
+  const [searchTerm, setSearchTerm] = useState<string>('')
+  const loadMoreRef = useRef(null)
+  const pageSize = useRef(20)
```
Comment (chetanvekariya-harness) [inline src/modules/70-pipeline/pages/execution-list/ExecutionListFilterForm/ExecutionListFilterForm.tsx:57]: Does this have to be ref?

**Review by chetanvekariya-harness — DISMISSED** [review]

File: src/modules/70-pipeline/pages/execution-list/ExecutionListFilterForm/ExecutionListFilterForm.tsx (hunk 4 above)
Comment (satyam-harness) [reply src/modules/70-pipeline/pages/execution-list/ExecutionListFilterForm/ExecutionListFilterForm.tsx:57]: Yes, we are  attaching this ref to last element of the list, and when that comes in viewport making next call to fetch more services

File: src/modules/70-pipeline/pages/execution-list/ExecutionListFilterForm/ExecutionListFilterForm.tsx (hunk 2 above)
Comment (satyam-harness) [reply src/modules/70-pipeline/pages/execution-list/ExecutionListFilterForm/ExecutionListFilterForm.tsx:212]: ohh yes

**Pull request closed without merging**